src/controllers/organizationController.ts

```diff
@@ -196,61 +209,134 @@ export default class OrganizationController {
   }
 
+  private async deleteAssociatedPreDraftOrganization(
+    userEmail: string
+  ): Promise<Option<IResponseErrorInternal | IResponseErrorConflict>> {
+    const errorOrMaybeOrganizationInstance = await getOrganizationInstanceFromDelegateEmail(
+      userEmail
+    );
+    if (errorOrMaybeOrganizationInstance.isLeft()) {
+      log.error(
+        "An error occurred reading data from db. %s",
+        errorOrMaybeOrganizationInstance.value
+      );
+      return some(
+        ResponseErrorInternal("An error occurred reading data from DB")
+      );
+    }
+    const organizationInstance = errorOrMaybeOrganizationInstance.value.toNullable();
+    if (organizationInstance) {
+      if (
+        organizationInstance.registrationStatus !==
+        OrganizationRegistrationStatusEnum.PRE_DRAFT
+      ) {
+        return some(
+          ResponseErrorConflict(
+            "There is already a registered organization associated to your account"
+          )
+        );
+      } else {
+        // The organization already associated to the user
+        // is still in draft or pre-draft status,
+        // so its registration process must be canceled
+        const maybeError = await deleteOrganization(organizationInstance);
+        if (maybeError.isSome()) {
+          log.error(
+            `An error occurred when canceling the registration process for the organization ${organizationInstance.ipaCode}. %s`,
+            maybeError.value
+          );
+          return some(
+            ResponseErrorInternal(
+              `An error occurred when canceling the previous registration process for the organization ${organizationInstance.ipaCode}`
+            )
+          );
+        }
+      }
+    }
+    return none;
+  }
+
+  private async signAndSendDocuments(
+    delegateFiscalCode: FiscalCode,
+    organizationInstance: OrganizationModel
+  ): Promise<IResponseErrorInternal | IResponseNoContent> {
+    const unsignedContractPath = `./documents/${organizationInstance.ipaCode}/contract.pdf`;
+    const signedContractPath = `./documents/${organizationInstance.ipaCode}/signed-contract.pdf`;
+    const unsignedMandatePath = `./documents/${
+      organizationInstance.ipaCode
+    }/mandate-${delegateFiscalCode.toLocaleLowerCase()}.pdf`;
+    const signedMandatePath = `./documents/${
+      organizationInstance.ipaCode
+    }/signed-mandate-${delegateFiscalCode.toLocaleLowerCase()}.pdf`;
+
+    const arrayOfMaybeError = await Promise.all([
+      this.createSignedDocument(unsignedContractPath, signedContractPath),
+      this.createSignedDocument(unsignedMandatePath, signedMandatePath)
+    ]);
+    const errorsArray = arrayOfMaybeError.filter(isSome);
+    errorsArray.forEach(error =>
+      log.error("An error occurred while signing documents. %s", error)
+    );
+    if (errorsArray.length > 0) {
+      return ResponseErrorInternal("An error occurred while signing documents");
+    }
+    try {
+      await organizationInstance.update({
```

Comment:
not needed by this pr but avoid try / catch blocks, use [TaskEither.tryCatch](https://github.com/gcanti/fp-ts/blob/1.12.0/src/TaskEither.ts#L283) instead to wrap code that may throw, ie.

`tryCatch(() => organizationInstance.update({...}).map(...)`

or an assigment if you really need early returns:

`const errorOrResult = tryCatch(() => organizationInstance.update({...})`